testing-overhead/Dockerfile-petclinic-base

```diff
@@ -0,0 +1,29 @@
+FROM adoptopenjdk:11-jdk
+
+# This is the base image that will contain a built version of the spring-petclinic-rest
+# application. Installing the dependencies and maven compiling the application is time
+# consuming, so we do it in a base image to save time nightly.
+
+# Allow the test runner to ssh into our container to start and control
+# the application under test.
+
+RUN apt update && \
+    apt install -y openssh-server sudo git && \
+    mkdir /run/sshd && \
+    sed -i 's/^#PasswordAuthentication.*/PasswordAuthentication yes/' /etc/ssh/sshd_config && \
+    sed -i 's/^#PermitUserEnvironment.*/PermitUserEnvironment yes/' /etc/ssh/sshd_config && \
+    sed -i 's/^#PermitEmptyPasswords.*/PermitEmptyPasswords yes/' /etc/ssh/sshd_config && \
+    sed -i 's/secure_path="/secure_path="\/opt\/java\/openjdk\/bin:/' /etc/sudoers && \
+    sed -i 's/PATH="/PATH="\/opt\/java\/openjdk\/bin:/' /etc/environment && \
+    useradd -m -d /app -s /bin/bash -u 5001 remote && \
+    passwd -d remote && \
+    mkdir -p /app/.ssh && chmod 700 /app/.ssh && \
+    ln -s /etc/environment /app/.ssh/environment && \
+    echo "%remote ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+
+WORKDIR /app
+RUN git clone http://github.com/spring-petclinic/spring-petclinic-rest.git
+WORKDIR /app/spring-petclinic-rest
+
+RUN ./mvnw package -Dmaven.test.skip=true
```

Comment:
That is one more argument in favour of microservice version: it already can build itself into docker images

Reply:
This way will leave the full maven local repo inside the container

Reply:
Does it really matter that much? Would you prefer that I remove the source code and just keep the build output jar?

Reply:
If we don't plan to share this image outside the job - it does not matter. If we do, it does :)

Reply:
You could also use a multi-stage dockerfile to avoid keeping the trash produced by the build (and build tools themselves) in the image.

Reply:
Ok, reworked to be a multi-stage dockerfile that no longer leaves the source tree behind. Looks like the image size goes from 839675046 to 636291599, so I guess that's a good thing.